metldata/load/main.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Compose the main app and resolve dependencies."""
+
+from functools import partial
+
+from fastapi import FastAPI
+from ghga_service_commons.api import configure_app
+from hexkit.protocols.dao import DaoFactoryProtocol
+from hexkit.providers.mongodb import MongoDbConfig
+from motor.motor_asyncio import AsyncIOMotorClient
+
+from metldata.load.api import rest_api_factory
+from metldata.load.config import ArtifactLoaderAPIConfig
+
+
+async def clear_database(config: MongoDbConfig):
+    """Clear a mongodb database."""
+
+    client = AsyncIOMotorClient(config.db_connection_str.get_secret_value())
+    db = client[config.db_name]
```

Comment:
I am not sure because the database might be preconfigured by the helm charts and the service receives credentials for that database. Deleting that database might mean that the service looses access to the MongoDB server?

Reply:
Also, the database user might/should not have the rights to drop a database. Probably better to leave it as is.